oauth-common/src/main/java/io/strimzi/kafka/oauth/services/ValidatorKey.java

```diff
@@ -58,6 +64,24 @@
         this.hasHostnameVerifier = hasHostnameVerifier;
         this.connectTimeout = connectTimeout;
         this.readTimeout = readTimeout;
+        this.enableMetrics = enableMetrics;
+
+        this.configIdHash = IOUtil.hashForObjects(validIssuerUri,
```

Comment:
I'm a bit puzzled what is this used for and how it relates to the metrics. Can you elaborate on it?

Reply:
When there are many listeners configured it results in multiple instances of the validator callback handler and consequently of the validators. I introduced an `oauth.config.id` config parameter to allow the user to set the name for the listener - it does not seem to be possible to get the listener name from the custom ValidateCallbackHandler. You want to segment the metrics by validators (listeners) rather than have it all merged together. And you need a discriminator (a context id) to qualify metrics. Thus if `oauth.config.id` was not set, it is calculated from other config parameters, resulting in a short hashstring so that every validator has a config id that is included as an attribute in the metric. When you map this to Prometheus JMX Exporter, you can then do PromQL queries on a per validator basis or you automatically get segmentation on a per validator basis.

Reply:
Could we use `Objects.hash()` instead of all this custom code?